nodeserver/controllers/routes.js

```diff
@@ -0,0 +1,82 @@
+const fs = require('fs');
+const path = require('path');
+const http = require('http');
+const bodyParser = require('body-parser');
+
+module.exports = (server) => {
+
+  server.use(bodyParser.urlencoded( { extended: true } ));
+
+  /**
+   * Controller that routes requests based on the URL using ExpressJS.
+   * Server runs on PORT 3000.
+   **/
+
+  // The root of the project is index.html, which is opened when the project is opened.
+  server.get('/', (request, response) => {
+    response.sendFile(path.resolve('./public/index.html'));
+  });
+
+  /** This API call can be used to get the current data.
+   * Reads data from JSON file and returns it.
+   * Using a JSON file since the data is small. Can be extended to DB.
+   **/
+  server.get('/publications/', (request, response) => {
+    const fileContent = fs.readFileSync('data/publications.json');
```

Comment:
Not important to fix here since speed is not a concern, but in general you want to avoid using `readFileSync` or any other `Sync`hronous functions in NodeJS - it blocks *all* other requests to the server during the time that Node is reading the file since NodeJS is single-threaded. It would be better to use the async `readFile`.